frontend/js/src/utils/utils.tsx

```diff
@@ -1058,6 +1058,16 @@ export function getPersonalRecommendationEventContent(
   );
 }
 
+export function getObjectForURLSearchParams(
+  urlSearchParams: URLSearchParams
+): Record<string, string> {
+  const object: Record<string, string> = {};
+  urlSearchParams.forEach((value, key) => {
+    object[key] = value;
+  });
+  return object;
```

Comment:
Not sure how relevant, but if you like one-liners:

```suggestion
  return Object.fromEntries(urlSearchParams);
```

Reply:
My code editor is throwing an error 
```Property 'fromEntries' does not exist on type 'ObjectConstructor'. Do you need to change your target library? Try changing the 'lib' compiler option to 'es2019' or later.ts(2550)
```
Strange. We're using `es2022`

Reply:
We're using es2022 _modules_ definition, our compilation target is es5, and this would still be a separate configuration option under `lib` : https://www.typescriptlang.org/tsconfig#lib

Forget I said anything, your solution is fine  :sweat_smile: